### PR TITLE
Refactor sections in RzBinPlugin from list to pvector

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

CI is green.

**Closing issues**

partially address https://github.com/rizinorg/rizin/issues/3755
partially address https://github.com/rizinorg/rizin/issues/3775

### CODE REVIEW

File: librz/util/vector.c [hunk 1]
```diff
@@ -444,6 +444,29 @@ RZ_API RZ_BORROW void **rz_pvector_find(RZ_NONNULL const RzPVector *vec, RZ_NONN
 	return NULL;
 }
 
+/**
+ * \brief Joins 2 pvector into one (pvec2 pointer needs to be freed by the user)
+ *
+ **/
+
```
Comment (XVilka) [inline librz/util/vector.c]: Unnecessary empty line

File: librz/util/vector.c [hunk 2]
```diff
@@ -444,6 +444,29 @@ RZ_API RZ_BORROW void **rz_pvector_find(RZ_NONNULL const RzPVector *vec, RZ_NONN
...
+RZ_API bool rz_pvector_join(RZ_NONNULL RzPVector *pvec1, RZ_NONNULL RzPVector *pvec2) {
+	rz_return_val_if_fail(pvec1 && pvec2, 0);
+
+	if (rz_pvector_empty(pvec2)) {
+		return false;
+	}
+
+	void **it;
+	rz_pvector_foreach (pvec2, it) {
+		rz_pvector_push(pvec1, *it);
```
Comment (XVilka) [inline librz/util/vector.c]: Can't we use `memmove` to speed up this somehow? Both are contiguous chunks of memory.

**Review by XVilka — CHANGES_REQUESTED** [review]
LGTM with two small nitpicks

File: librz/util/vector.c (hunk 2 above)
Comment (XVilka) [reply librz/util/vector.c]: For example, calling `RESIZE_OR_RETURN_NULL` first, then `memmove` might be faster than iterations if there are many elements in `pvec2`.

**Review by wargio — APPROVED** [review]

**Pull request merged**